File: backend/src/main/java/t/lab/guide/dto/excursion/ExcursionListResponse.java
package t.lab.guide.dto.excursion;

import io.swagger.v3.oas.annotations.media.ArraySchema;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;

import java.util.List;

@Schema(description = "Ответ с краткой информацией о найденных экскурсиях")
@Builder
public record ExcursionListResponse(
        @ArraySchema(
                arraySchema = @Schema(description = "Список экскурсий"),
                schema = @Schema(implementation = ExcursionShortItem.class)
        )
        List<ExcursionShortItem> excursions
) {}

File: backend/src/main/kotlin/t/lab/guide/dto/excursion/ExcursionListResponse.kt
package t.lab.guide.dto.excursion

import io.swagger.v3.oas.annotations.media.ArraySchema
import io.swagger.v3.oas.annotations.media.Schema

@Schema(description = "Ответ с краткой информацией о найденных экскурсиях")
data class ExcursionListResponse(
    @ArraySchema(
        arraySchema = Schema(description = "Список экскурсий"),
        schema = Schema(implementation = ExcursionShortItem::class),
    )
    val excursions: List<ExcursionShortItem>,
)
